evo_researcher/functions/web_search.py

```diff
@@ -1,9 +1,14 @@
 import os
+import tenacity
 from dotenv import load_dotenv
 from tavily import TavilyClient
 
 from evo_researcher.models.WebSearchResult import WebSearchResult
+from evo_researcher.functions.cache import persisted_inmemory_cache
 
+
+@tenacity.retry(stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_fixed(1), reraise=True)
+@persisted_inmemory_cache
```

Comment:
This is great for robustness and cheapness of benchmarking (fewer api calls), but one side-affect of this is that it will distort the timing results of the agent that calls this function. Do you know if there's a way to mitigate this? e.g. if there's a cache hit, wait some amount of time, which is roughly the same as the length of a `web_search` call

Reply:
Hmm, I didn't think about this before 🤔 I could add some logic about sleeping if cache is hit, but maybe just running benchmark with `DISABLE_CACHE=1 python ...` is enough?  (when we want to compare timing)

Because I'd say the timing is not that important, if agents are making ~30 predictions per day, it can take 30min per prediction and it's still okay, as long as it is cost effective of course.

Reply:
Hmm, I think either:
1. we decide timing is not an important metric and remove it from the benchmark
2. try to simulate the timing while using the cache
3. we don't try to simulate the timing, but add a warning to the benchmark report if caching is enabled
4. in the benchmark, run all markets with caching enabled, then re-run a sample with it disabled, and use that for timing.

I reckon (2) or (3). wbu?

Reply:
I like (3).

(2) might get complicated and erroneous in the end anyway.

I'll add that 👌